### DISCUSSION TITLE
Need help writing custom plugins for Blender Malt

### DISCUSSION THREAD

**Discussion by master-wayne** [bnpr/Malt#489] [Q&A] [answered]
Hello!
I am new to Blender Malt and getting myself acquainted with writing simple custom glsl shaders. My issue is similar to the one mentioned [here](https://github.com/bnpr/Malt/discussions/237#discussion-3860099).
I cannot get the instance_random_color() function to show as node in the Malt Node Tree. I have setup the global Plugins folder as mentioned in the thread and placed the PluginExample.glsl file directly under it.

Here are the steps that I have tried:

1. Define Global Malt Plugin directory:
   ......Blender/blender-3.4.1-windows-x64/Malt/

2. Download the sample PluginExample.glsl from here: [PluginExample/Shaders]
(https://github.com/bnpr/Malt/tree/Development/plugins/PluginExample/Shaders)

3. Paste the sample file under Global Malt Plugin directory.

4. Restart Blender / Reaload Malt pipeline from Pipeline settings.

![image](https://user-images.githubusercontent.com/18322773/225710330-41eaa4d8-7915-482d-a9e8-55abcc5ee303.png)

I have gone through the [online documentation](https://malt3d.com/documentation/getting%20started/) and various support threads here but could not find a solution. My ultimate goal is to write a custom node to perform Stencil Testing to hide parts of a character's hair but it seems it is not possible with the nodes currently available in the Malt Node Tree.

**Comment by pragma37** [comment, marked as answer]
The plugin is the **whole** `PluginExample` **folder**, not just the `PluginExample.glsl` file.
So you need to drop the `Plugins Example` folder into your plugin directory.

As a side note, I would move the Global Plugin directory outside your Blender installation, so you don't accidentally delete it when installing a new Blender version.

**Comment by master-wayne** [comment]
Hi
Thanks, that worked!
![image](https://user-images.githubusercontent.com/18322773/225817879-f7b80428-2ee2-47a0-9d68-795b3a5ca443.png)

Also thanks for pointing out the location of the Plugin directory, I will change it accordingly.
